### DISCUSSION TITLE
Preserving previous state when going back to loading in StreamProvider

### DISCUSSION THREAD

**Discussion by lockieRichter** [rrousselGit/riverpod#2471] [Q&A] [answered]
Hi,

I see in the changelog for version 2.1.0 there is now the ability to preserve previous state when going back to loading in `StreamProvider`.

> FutureProvider, StreamProvider and AsyncNotifierProvider now preserve the previous data/error when going back to loading. This is done by allowing AsyncLoading to optionally contain a value/error.

Is anyone able to show me an example of how this is achieved for a `StreamProvider`, I can't quite figure out what I need to do!

Thanks!

**Comment by rrousselGit** [comment, marked as answer]
You don't need to do anything special. That's the default

**Reply by lockieRichter** [reply]
Ok great news, thanks for the swift reply.
